plugins/com.google.cloud.tools.eclipse.projectselector/src/com/google/cloud/tools/eclipse/projectselector/ProjectRepository.java

```diff
@@ -103,7 +96,32 @@ private static Projects getProjectsApi(Credential credential) {
   }
 
   private static GcpProject convertToGcpProject(Project project) {
+    Preconditions.checkNotNull(project);
     return new GcpProject(project.getName(), project.getProjectId());
   }
 
+  /**
+   * @return true if the credential has access to the GCP project identified by {@code projectId}
+   * and the project has App Engine application
+   * @throws ProjectRepositoryException if an error other than HTTP 404 happens while retrieving the
+   * App Engine application
+   */
+  public boolean hasAppEngineApplication(Credential credential, String projectId)
```

Comment:
Let's figure out who to file that RFE against then. If I'm reading this right we could be making hundreds of network class, just for opening a dialog. That makes me very nervous.